docs/gateway-deployment-via-docker-compose.md

```diff
@@ -0,0 +1,43 @@
+# Deploy Gateway via docker-compose using local servers
+
+* The docker-compose.yml for proxies should work out of the box for usual CoatRack user who seeks to connect with a remote Web Application and a Config Server. 
+* However, as developers you will set up local servers. The problem is that when a docker container accesses `localhost`, it will send a request to itself instead of a host device port. To realize this setup anyhow, perform the following instructions.
+
+
+
+### Starting the Config Server
+
+* Remove the the following attributes from Config Server application.yml and start the Config Server.
+
+  ```yml
+  spring:
+    datasource <- 
+    jpa:
+      properties:
+        hibernate:
+          dialect <- 
+  ```
+
+  
+
+### Configuring  Admin/Web Application
+
+* `localhost` needs to be replaced or overridden with `host.docker.internal` (see attributes below) and start the admin server. For that use a custom profile like application-private-\<name\>.yml and override these parameters in the application.yml.
```

Comment:
It will be quite cumbersome for every developer has to follow these instructions to get a working local setup. To make it easier, I would suggest one of the following options (to be discussed which one to go for):

- we could either include these instructions directly in our template for "private" config files, so that there is no need to copy/paste into that file
- or we could create specific non-private profiles to support these different settings.

Reply:
I would not put these instructions to our template file, as aside from this feature, we will need `localhost` and not `host.docker.internal`. Therefore we should go with the second approach.

Reply:
Done. I introduced the profile `issue-170-test-setup`. The extra documentation became redundant and was therefore deleted. Testing this feature should be much simpler now. But it is still important to start the config server with an H2 DB, not with the Postgres configuration.